calico_containers/pycalico/datastore.py

```diff
@@ -745,6 +764,46 @@ def update_endpoint(self, endpoint):
         endpoint._original_json = new_json
 
     @handle_errors
+    def create_endpoint(self, hostname, orchestrator_id, workload_id,
+                        ip_list, mac=None):
+        """
+        Create a single new endpoint object.
+
+        Note, the endpoint will not be stored in ETCD until set_endpoint
+        or update_endpoint is called.
+
+        :param hostname: The hostname that the endpoint lives on.
+        :param orchestrator_id: The workload that the endpoint belongs to.
+        :param workload_id: The workload that the endpoint belongs to.
+        :param ip_list: A list of ip addresses that the endpoint belongs to
+        :param mac: The mac address that the endpoint belongs to
+        :return: An Endpoint Object
+        """
+        ep = Endpoint(hostname=hostname,
+                      orchestrator_id=orchestrator_id,
+                      workload_id=workload_id,
+                      endpoint_id=uuid.uuid1().hex,
+                      state="active",
+                      mac=mac)
+        next_hops = self.get_default_next_hops(hostname)
+
+        for ip in ip_list:
+            try:
+                next_hop = next_hops[ip.version]
+            except KeyError:
+                raise AddrFormatError("This node is not configured for IPv%d."
+                                       % ip.version)
+            network = IPNetwork(IPAddress(ip))
```

Comment:
No need to wrap IP in IPAddress().  You're already accessing ip.version above, so ip should already be an IPAddress object.

Maybe worth asserting this is the case.